sqllineage/core/parser/sqlfluff/utils.py

```diff
@@ -413,6 +413,13 @@ def is_union(segment: BaseSegment) -> bool:
     )
 
 
+def is_union_direct(segment: BaseSegment) -> bool:
+    """
+    like is_union but use segments rather than raw_segments
+    """
+    return len([s for s in segment.segments if s.type == "set_expression"]) > 0
+
+
```

Comment:
Any particular reason we had to add this. The test case would also pass using `is_union`